kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerInstanceManager.java

```diff
@@ -467,9 +462,14 @@ public boolean isAlive(ServerInstanceKey serverInstanceKey) {
         return alive;
     }
 
-    protected KieServicesClient getClient(String url) {
-        KieServicesClient client = clientProviders.stream().filter(provider -> provider.supports(url)).findFirst().get().get(url);
-        logger.debug("Using client {}", client);
+    protected KieServicesClient getClient(final String url) {
+        KieServicesClientProvider clientProvider = clientProviders.stream().filter(provider -> provider.supports(url)).findFirst().orElseThrow(() -> new KieServerControllerNotFoundException("Kie Services Client Provider not found for url: " + url));
+        logger.debug("Using client provider {}", clientProvider);
+        KieServicesClient client = clientProvider.get(url);
+        logger.debug("Using client {}", clientProvider);
```

Comment:
should be client instead of clientProvider

Reply:
Done